### ISSUE TITLE
Indicators As Features

### ISSUE THREAD

**Issue by bitnom** [philipxjm/Neural-Stock-Market-Prediction#3]
What if, instead of simply using the price history, we included indicators such as RSI, MACD, Polarized Fractal Efficiency, etc? Should that not improve the accuracy? I'd like to investigate ways of incorporating such extra features. Any thoughts on accomplishing this?

**Comment by amirkhango** [comment]
I am very glad to find somebody focusing on Stock Market with DL technique. Before finding this github, I also pay some time on this and beat with some problems, this code gives me a good reference.
  **Unfortunately**, I find a fatal bug that indicates this github's CNN codes **cannot run in reality**. That is
```
# shuffling the data
perm = np.arange(labels_set.shape[0])
np.random.shuffle(perm)
stocks_set = stocks_set[perm]
labels_set = labels_set[perm]
```

  The reason for that, in reality, stock stream comes with time order, this shuffle operation will break this rule.
  If you do not believe, you can comment these codes and run it, you would find your test accuracy dramatically decrease from 70% down to 50%. That means it can only behave well on train set but always sucks on test set.

 TL;DR, stock price/index prediction by CNN is nearly **a hard (even impossible) task**, because it is essentially **event-driven** not data-driven (price-driven). The codes here belongs to data-driven (price-driven) methods.
 By the way, some researchers now pay much more attention on **event-driven** method, i.e., they grabbed event data of daily news and then use Deep Learning model to analysis it. But this method is relatively complicated.

**Comment by Leungfranklin** [comment]
I once have the same thoughts on adding RSI etc. However, if CNN can recognize patterns, I suppose it can also recognize any momentum patterns. If the accuracy is actually around 50%, that means there are no momentum patterns at all in price movement :-)

**Comment by dboe1107** [comment]
But isn't the window +5, so that would be treated as a block of "time" records.  Doesn't that force the time sequence pattern?
